Lib/glyphsLib/builder/anchors.py

```diff
@@ -132,17 +132,17 @@ def _adjust_anchors(anchor_data, ufo, parent, component):
     t = Transform(*component.transformation)
     _componentAnchor = _componentAnchorFromLib(parent, component)
     for anchor in glyph.anchors:
-        # adjust either if this anchor has data and the component also contains
+        # adjust either if component is attached to a specific named anchor
+        # (e.g. top_2 for a ligature glyph)
+        # rather than to the standard anchors (top/bottom)
+        if _componentAnchor in anchor_data:
```

Comment:
I know it's unlikely that `anchor_data` dict contains an item with key `None` or `""`, but to avoid surprises it'd be better to check `if _componentAnchor and _componentAnchor in anchor_data`

Reply:
Okay, done